src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -567,6 +568,11 @@ public DatabaseRecord ReadDatabase(TransactionOperationContext context, string n
             var doc = Read(context, "db/" + name.ToLowerInvariant(), out etag);
             if (doc == null)
                 return null;
+            // If we are in passive state, we have prevent from tasks to be performed by this node.
+            if (_parent.CurrentState == RachisConsensus.State.Passive)
```

Comment:
This block the access to the database state, not the database itself.
I'll still be able to access the storage of the data, indexes, etc.